### PR TITLE
Add documentation to CA and CPP

### PR DESCRIPTION
**Pull request by hanak**
**Purpose**
Based on the recent pull request I realized I did not include `CA()` an `CP()` (which I added recently) to `api.rst`. Therefore, this pull request should fill the gap.

**Solution**
1. Add `CA()` and `CP()` to `api.rst` including an example.
2. Move Python comments into codedoc style comments for specifiers such `CA()`. This allows IDEs (such as VSCode) to show hints.
3. Add `NP()` to `test/examples.py`

**Test**
1. No code change. `api.rst` tested in VSCode preview.

### CODE REVIEW

File: epicsdbbuilder/recordbase.py [hunk 1]
```diff
@@ -304,34 +304,95 @@ def Value(self):
...
+
+    Example (Python source)
+    -----------------------
+    `my_record.INP = NP(other_record)`
+
+    Example (Generated DB)
+    ----------------------
+    `field(INP, "other NPP")`
+    """
     return record('NPP')
```
Comment (Araneidae) [inline epicsdbbuilder/recordbase.py:396]: Can now lose these lines: we've done them all now (I think).

**Comment by Araneidae** [comment]
Thanks.  I did spot this the other day while reviewing your other changes, but forgot to do anything about it.

File: epicsdbbuilder/recordbase.py (hunk 1 above)
Comment (Araneidae) [reply epicsdbbuilder/recordbase.py:396]: Eh?  Those weren't the lines I selected!  I meant to lose the last three lines in the file.  I guess github is just showing the closest patch.

File: epicsdbbuilder/recordbase.py [hunk 2]
```diff
@@ -304,34 +304,95 @@ def Value(self):
...
 
 # ... put the rest in some time
```
Comment (Araneidae) [inline epicsdbbuilder/recordbase.py:398]: These lines?

Comment (hanak) [reply epicsdbbuilder/recordbase.py]: Yes. Yet, we miss `NMS`, `MSS` and `MSI` to become complete. I have added those.

**Comment by hanak** [comment]
As mentioned, I have added `NMS`, `MSS`, and `MSI`. I have used documentation and partial EPICS base source code search while compiling the documentation. I have not tried to run such modified links with EPICS.

**Pull request merged**